test/merge-patient-lab-iips-mocha.js

```diff
@@ -231,5 +306,143 @@ describe('merge-patient-lab-iips', function() {
                 payload.data.date.should.equal('2012-02-01');
              });
         });
+
+        it( "should handle JSON parse errors", function() {
+            var node = test.createComponent(compFactory);
+            sinon.stub( console, 'error' );
+            return new Promise(function(done, fail){
+
+                test.onOutPortData(node, 'error', fail);
+                test.onOutPortData(node, 'output', done);
+
+                test.sendData( node, 'patient',
+                              '{"id": "001", "name": "David", "dob": "1959-01-23" }');
+
+                // Note that this labwork record is not JSON - id is a mess and
+                // should cause a json parse error to be thrown
+                test.sendData( node,'labwork',
+                               '{"id 001, "glucose": "85", "date": "2013-02-01" }');
+
+             }).then(function(payload){
+
+                console.error.restore();
+                payload.should.exist;
+                payload.should.not.be.empty;
+                payload.should.have.all.keys( 'vnid', 'data', 'error', 'lm' );
+                payload.vnid.should.equal('');
+                payload.data.startsWith('Error: Unable to parse parameter').should.be.true;
```

Comment:
I don't think the output state data should contain the error message. I think it should be undefined in this case.